packages/ihe-gateway-sdk/src/models/shared.ts

```diff
@@ -16,55 +16,149 @@ export const oidStringSchema = z
+  newDocumentUniqueId: z.string().nullish(),
+  contentType: z.string().nullish(),
+  url: z.string().nullish(), // signed urls that mirth will use to download actually b64 bytes
+  uri: z.string().nullish(),
+  creation: z.string().nullish(),
+  title: z.string().nullish(),
 });
 
-export type DocumentReference = z.infer<typeof documentReference>;
+export type DocumentReference = z.infer<typeof documentReferenceSchema>;
 
-export const issue = z.object({
+export const codeSchema = z.object({
+  system: z.string(),
+  code: z.string(),
+});
+export type Code = z.infer<typeof codeSchema>;
+
+const detailsSchema = z.union([
+  z.object({ coding: z.array(codeSchema) }),
+  z.object({ text: z.string() }),
+]);
+
+export const issueSchema = z.object({
   severity: z.string(),
   code: z.string(),
-  details: z.object({ text: z.string() }),
+  details: detailsSchema,
 });
 
 export const operationOutcome = z.object({
   resourceType: z.string(),
   id: z.string(),
-  issue: z.array(issue),
+  issue: z.array(issueSchema),
 });
 
+export type OperationOutcome = z.infer<typeof operationOutcome>;
+
+export const xcpdPatientIdSchema = z.object({
+  id: z.string(),
+  system: z.string(),
+});
+
+export type XCPDPatientId = z.infer<typeof xcpdPatientIdSchema>;
+
 export const baseResponseSchema = z.object({
   id: z.string(),
-  cxId: z.string(),
   timestamp: z.string(),
   responseTimestamp: z.string(),
-  xcpdPatientId: z.object({ id: z.string(), system: z.string() }).nullable(),
-  patientId: z.string(),
-  operationOutcome: operationOutcome.nullable(),
+  xcpdPatientId: xcpdPatientIdSchema.nullish(),
+  patientId: z.string().nullish(), // TODO should this not be nullish
+  operationOutcome: operationOutcome.nullish(),
+});
+
+export type BaseResponse = z.infer<typeof baseResponseSchema>;
+
+const identifierSchema = z.object({
+  system: z.string().optional(),
+  value: z.string().optional(),
+});
+
+const nameSchema = z.object({
+  family: z.string(),
+  given: z.array(z.string()),
+  prefix: z.array(z.string()).optional(),
+  suffix: z.array(z.string()).optional(),
+});
+
+const telecomSchema = z.object({
+  system: z
+    .union([
+      z.literal("phone"),
+      z.literal("fax"),
+      z.literal("email"),
+      z.literal("pager"),
+      z.literal("url"),
+      z.literal("sms"),
+      z.literal("other"),
+    ])
+    .optional(),
+  value: z.string().optional(),
+});
+
+const genderSchema = z
+  .union([z.literal("male"), z.literal("female"), z.literal("other"), z.literal("unknown")])
+  .optional();
+
+const addressSchema = z.object({
+  line: z.array(z.string()).optional(),
+  city: z.string().optional(),
+  state: z.string().optional(),
+  postalCode: z.string().optional(),
+  country: z.string().optional(),
+});
+
+const extensionSchema = z.object({
+  url: z.string(),
+  valueAdress: addressSchema.optional(),
+  valueString: z.string().optional(),
+});
+
+export const patientResourceSchema = z.object({
```

Comment:
Had to define all this since we are receiving patient resources and you cant go from the medplum fhir Patient type to zod :(